public/app/plugins/panel/timeseries/TimeSeriesTooltip.tsx

```diff
@@ -53,119 +127,47 @@ export const TimeSeriesTooltip = ({
   isPinned,
   annotate,
 }: TimeSeriesTooltipProps) => {
-  const theme = useTheme2();
   const styles = useStyles2(getStyles);
 
   const xField = seriesFrame.fields[0];
-  if (!xField) {
-    return null;
-  }
-
-  const xFieldFmt = xField.display || getDisplayProcessor({ field: xField, theme });
-  let xVal = xFieldFmt(xField!.values[dataIdxs[0]!]).text;
 
-  let links: Array<LinkModel<Field>> = [];
-  let contentLabelValue: LabelValue[] = [];
+  const xVal = xField.display!(xField.values[dataIdxs[0]!]).text;
```

Comment:
uPlot guarantees `dataIdxs[0]` to exist in `mode: 1`, since that's how it determines the starting point for finding hovered points. it's the closest index in the x field relative to cursor position.

we can always fall back to the first x value by doing `?? 0`. i'm not sure that's "reasonable", though, since it would just mask a legit bug we should fix.